File: main-service/src/main/java/ru/pgk/main_service/features/teacher/service/TeacherService.java
package ru.pgk.main_service.features.teacher.service;

import ru.pgk.main_service.features.teacher.dto.params.AddOrUpdateTeacherParams;
import ru.pgk.main_service.features.teacher.entities.TeacherEntity;

public interface TeacherService {
    TeacherEntity add(AddOrUpdateTeacherParams params);

    TeacherEntity update(Integer id, AddOrUpdateTeacherParams params);

    TeacherEntity deleteById(Integer id);
}

File: teacher-service/src/main/kotlin/ru/pgk/teacher_service/features/services/TeacherService.kt
package ru.pgk.teacher_service.features.services

import ru.pgk.teacher_service.features.dto.params.AddOrUpdateTeacherParams
import ru.pgk.teacher_service.features.entities.TeacherEntity

interface TeacherService {
    fun add(params: AddOrUpdateTeacherParams): TeacherEntity
    fun update(id: Int, params: AddOrUpdateTeacherParams): TeacherEntity
    fun deleteById(id: Int): TeacherEntity
}